aea/runtime.py

```diff
@@ -22,24 +22,40 @@
 
 import asyncio
 import logging
-from abc import ABC, abstractmethod
 from asyncio.events import AbstractEventLoop
+from concurrent.futures._base import CancelledError
 from contextlib import suppress
 from enum import Enum
 from typing import Dict, Optional, Type, cast
 
 from aea.abstract_agent import AbstractAgent
 from aea.agent_loop import AsyncAgentLoop, AsyncState, BaseAgentLoop, SyncAgentLoop
+from aea.connections.base import ConnectionStates
 from aea.decision_maker.base import DecisionMaker, DecisionMakerHandler
-from aea.helpers.async_utils import ensure_loop
+from aea.helpers.async_utils import Runnable
 from aea.helpers.exception_policy import ExceptionPolicyEnum
-from aea.multiplexer import AsyncMultiplexer, Multiplexer
+from aea.multiplexer import AsyncMultiplexer
 from aea.skills.tasks import TaskManager
 
 
 logger = logging.getLogger(__name__)
 
 
+class StopRuntime(Exception):
```

Comment:
It should be made explicit here that this is for internal use and how it is to be used @solarw (https://github.com/fetchai/agents-aea/pull/1731/files#r487774129). Future developers will then have it easier to use this.

Reply:
consider prepending `_`

Reply:
done